### PR TITLE
Remove fatal errors, instead logging and continuing

### LINKED ISSUE #23: Cut down on fatal errors
In azure.go, there are currently 15 uses of `log.Fatalf`. Which of them are reasonable to keep as fatal? For example, partial configuration errors (such as pointing to non-existing resources, which gives a 404) will right now quit the exporter.
Some errors _might_ make sense to quit with (invalid credentials, perhaps), but most not. Happy to do the changes when there is agreement on what is best.

Ping @brian-brazil

### PR DESCRIPTION
**Pull request by carlpett**
Fixes #23.

### CODE REVIEW

File: azure.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func NewAzureClient() *AzureClient {
 	}
 }
 
-func (ac *AzureClient) getAccessToken() {
+func (ac *AzureClient) getAccessToken() error {
```
Comment (brian-brazil) [inline azure.go]: You're not checking the return value of this

Comment (carlpett) [reply azure.go]: Whoops, good catch. Fixing now.
It is called during startup too, here:
https://github.com/RobustPerception/azure_metrics_exporter/blob/17b15ae2bb28698cc1ed30d7cc27c918da1dc7b5/main.go#L118-L126

Should this be considered "configuration error", and thus fatal, or should we try to keep going anyway?

Comment (brian-brazil) [reply azure.go]: At startup that's a failure I'd say, as we can't work.

Comment (carlpett) [reply azure.go]: Agreed. Fixed.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**